test/Datadog.Trace.DuckTyping.Tests/DuckIgnoreTests.cs

```diff
@@ -0,0 +1,146 @@
+using Xunit;
+
+#pragma warning disable SA1201 // Elements must appear in the correct order
+#pragma warning disable SA1402 // File may only contain a single class
+
+namespace Datadog.Trace.DuckTyping.Tests
+{
+    public class DuckIgnoreTests
+    {
+        [Fact]
+        public void NonPublicStructCopyTest()
+        {
+            PrivateStruct instance = default;
+            CopyStruct copy = instance.DuckCast<CopyStruct>();
+            Assert.Equal((int)instance.Value, (int)copy.Value);
```

Comment:
Shouldn't you assign a value first? Here the assertion will work even if Value isn't actually copied to the proxy, since you're keeping the default value

Reply:
Never mind, I didn't realize it was a property